File: Dhiva/moviecatalogue-subfinal/app/src/main/java/id/dhiva/moviecatalogue/ui/movie/MovieViewModel.java
package id.dhiva.moviecatalogue.ui.movie;

import java.util.ArrayList;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;
import id.dhiva.moviecatalogue.data.DataRepository;
import id.dhiva.moviecatalogue.model.ModelItem;

public class MovieViewModel extends ViewModel {
    private MutableLiveData<ArrayList<ModelItem>> allMovieList = new MutableLiveData<>();
    private DataRepository dataRepository;

    public MovieViewModel(DataRepository dataRepository) {
        this.dataRepository = dataRepository;
    }

    MutableLiveData<ArrayList<ModelItem>> getAllMovieList(){
        allMovieList.postValue(dataRepository.getAllMovies());
        return allMovieList;
    }


}

File: FinalSubmission/app/src/main/java/cikal/dicoding/finalsubmission/ui/movie/MovieViewModel.kt
package cikal.dicoding.finalsubmission.ui.movie

import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import cikal.dicoding.finalsubmission.data.source.CatalogueRepository
import cikal.dicoding.finalsubmission.data.source.local.entity.MovieEntity

class MovieViewModel(private val catalogueRepository: CatalogueRepository) : ViewModel() {

    fun getMovies(): LiveData<List<MovieEntity>> = catalogueRepository.getAllMovies()
}